### ISSUE TITLE
add settings UI for auth endpoints

### ISSUE THREAD

**Issue by snarfed** [indieweb/wordpress-micropub#72]
...specifically, a way to opt into either wordpress's internal auth or a different auth/token server than indieauth. details in https://github.com/snarfed/wordpress-micropub/#authentication-and-authorization .

@miklb graciously volunteered to look into this. thank you! i've never done wordpress admin UI before, so i'm pretty open to ideas or direction here.

**Comment by miklb** [comment]
I'm looking at a plugin to do GitHub as a syndication target and think I would like to store the access token for a user in the db, and so thinking about where in the admin interface I should add the field to enter, and thought about this issue.

Think it would be best to have a full page for all micropub related settings at this point?

**Comment by miklb** [comment]
What I'm getting at is if I'm adding a setting for GitHub silo.pub access token I could go ahead and tackle this issue at the same time.

**Comment by miklb** [comment]
ping @snarfed

**Comment by snarfed** [comment]
sorry, didn't realize you were waiting on me here!

github syndication plugin sounds great! agreed that it should be a separate plugin, though, not part of this one.

as for settings in the admin UI, i don't have a strong preference. @dshanske and @pfefferle are more familiar with design practices there. happy to follow your and their lead.

**Comment by miklb** [comment]
Right, I totally meant the GitHub plugin would be separate, just need to decide whether a separate page for settings or if anyone had a thought where it could fit in an existing page.

I've seen a few comments from people confused about the facepiles setting being Discuss, so leaning towards a separate micropub page that other plugins could piggy back onto.

**Comment by snarfed** [comment]
> I've seen a few comments from people confused about the facepiles setting being Discuss, so leaning towards a separate micropub page that other plugins could piggy back onto.

hmm. maybe! there's lots of existing knowledge and best practices in the wordpress community about how to design plugin admin/settings UIs. i'd encourage you to look into that first! e.g. https://developer.wordpress.org/plugins/administration-menus/ and beyond.

**Comment by dshanske** [comment]
I built a stub for this into my current work and will be adding to it.

**Comment by dshanske** [comment]
Going to push this out of the goal of a new release.

**Comment by dshanske** [comment]
Resolved via #181
